packages/nvidia_nat_core/src/nat/front_ends/fastapi/fastapi_front_end_plugin_worker.py

```
@@ -1263,147 +1263,165 @@ class MCPToolInfo(BaseModel):
+            Uses the per-user workflow builder to resolve function groups and
+            applies the same MCP client inspection logic as the shared endpoint.
+            """
+            per_user_manager = next((sm for sm in self._session_managers if sm.is_workflow_per_user), None)
+            if per_user_manager is None:
+                raise HTTPException(status_code=400, detail="No per-user workflow is configured.")
+
+            try:
+                async with per_user_manager.session(user_id=user_id, http_connection=request) as session:
+                    mcp_clients_info = await _collect_mcp_client_tool_list(session.workflow.function_groups)
+                    return MCPClientToolListResponse(mcp_clients=mcp_clients_info)
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)